ocaml/xapi/helpers.ml

```diff
@@ -2041,11 +2041,26 @@ let update_ca_bundle =
           )
     )
 
+(** {pp_hash hash} outputs the hexadecimal representation of the {hash}
+  adding a semicolon between every octet, in uppercase.
```

Comment:
Good point RE: (semi)colon, I just cut-pasted haha

RE: Digest.hex_of_string, it looks like it wasn't added until Ocaml 5.2.0, before which it could only convert exactly 16 characters. Hex.(show @@ of_cstruct hash) is doing this anyway, it's just the colon part that's finnicky, as Colin says

Reply:
```
let hex str =
  let hex = "0123456789ABCDEF" in
  let n = String.length str in
  assert (n > 0);
  String.init (n * 2 + (n-1)) (function
  | i when i mod 3 = 0 -> hex.[Char.code str.[i/3] lsr 4]
  | i when i mod 3 = 1 -> hex.[Char.code str.[i/3] land 0x0f]
  | _ -> ':')

```

Not sure this is much better; it's compact and relatively efficient.